File: capacitive/sim/calc_capacitance.m
k=1;
e=8.854e-12;
z=[0.1e-3,0.2e-3,0.5e-3]

wavenumber=6;
l=5.0e-3;
W=10.0e-3;
g=0.16e-3;

R=10e+6;

width_ratio = 1;

A_tx=wavenumber*W*l/8;
A_rx=(width_ratio*W)*(wavenumber)*l;

C_tx = k*e*A_tx./z
C_rx = k*e*A_rx./z

C_tx_comp = 1./(1./C_tx+1./(3*C_tx+C_rx))
C_rx_comp = 1./(1./C_rx+1./(4*C_tx))

% fc = 1./(R*C_rx_comp)./(2*pi)
attenuation = ((2*pi*1e+3)*R*C_rx_comp)./sqrt(((2*pi*1e+3)*R*C_rx_comp).^2+1)

File: capacitive/padgen/calc_capacitance.py
import numpy as np

k=1
e=8.854e-12
z=np.array([0.1e-3,0.2e-3,0.5e-3])

from pad_config import *

R=10e+6

A_tx=wavenumber*(1.0e-3*W)*(1.0e-3*l)/8
A_rx=(width_ratio*(1.0e-3*W))*(wavenumber)*(1.0e-3*(l-g))

C_tx = k*e*A_tx/z
C_rx = k*e*A_rx/z

# add in loading effect of other capacitors
C_tx_comp = 1/(1/C_tx+1/(3*C_tx+C_rx))
C_rx_comp = 1/(1/C_rx+1/(4*C_tx))

print("C_tx:", C_tx_comp)
print("C_rx:", C_rx_comp)

# fc = 1./(R*C_rx_comp)/(2*np.pi)
attenuation = ((2*np.pi*1e+3)*R*C_rx_comp)/np.sqrt(((2*np.pi*1e+3)*R*C_rx_comp)**2+1)
print("rx attenuation:", attenuation)
